Mathlib/Tactic/ToAdditive.lean

```diff
@@ -0,0 +1,467 @@
+    getParam := fun decl stx => do
+        let ids ← match stx with
+          | `(attr|to_additive_ignore_args $[$ids:num]*) => pure <| ids.map (·.isNatLit?.get!)
+          | _ => throwError "unexpected to_additive_ignore_args syntax {stx}"
+        return ids.toList
+  }
+
+/-- Gets the set of arguments that should be ignored for the given name (according to to_additive_ignore_args) -/
+def ignore [Functor M] [MonadEnv M]: Name → M (Option (List Nat))
+  | n => (ignoreArgsAttr.getParam · n) <$> getEnv
+
+initialize reorderAttr : ParametricAttribute (List Nat) ←
+  registerParametricAttribute {
+    name := `to_additive_reorder
+    descr := "Auxiliary attribute for `to_additive` that stores arguments that need to be reordered."
+    getParam := fun decl stx =>
+      match stx with
+      | `(attr|to_additive_reorder $[$ids:num]*) => pure <| Array.toList <| ids.map (·.isNatLit?.get!)
+      | _ => throwError "unexpected to_additive_reorder syntax {stx}"
+  }
+
+def getReorder [Functor M] [MonadEnv M]: Name →  M (List Nat)
+  | n => (reorderAttr.getParam · n |>.getD []) <$> getEnv
+
+def shouldReorder [Functor M] [MonadEnv M]: Name → Nat → M Bool
+  | n, i => (i ∈ ·) <$> (getReorder n)
+
+
+initialize relevantArgAttr : ParametricAttribute (Nat) ←
+  registerParametricAttribute {
+    name := `to_additive_relevant_arg
+    descr := "Auxiliary attribute for `to_additive` stating which arguments are the types with a multiplicative structure."
+    getParam := fun decl stx =>
+      match stx with
+      | `(attr|to_additive_relevant_arg $id) => pure <| id.isNatLit?.get!
+      | _ => throwError "unexpected to_additive_relevant_arg syntax {stx}"
+  }
+
+  def isRelevant [Functor M] [MonadEnv M]: Name → Nat → M Bool
+  | n, i =>
+    (fun | some j => i == j | none => i == 0)
+    <$> (relevantArgAttr.getParam · n)
+    <$> getEnv
+
+/- Maps multiplicative names to their additive counterparts. -/
+initialize translations : SimplePersistentEnvExtension (Name × Name) (NameMap Name) ←
+  registerSimplePersistentEnvExtension {
+    name          := `translations,
+    addImportedFn := fun ass => ass.foldl (init := ∅) fun
+                       | names, as => as.foldl (init := names) fun
+                         | names, (a,b) => names.insert a b,
+    addEntryFn    := fun s n => s.insert n.1 n.2 ,
+    toArrayFn     := fun es => es.toArray
+  }
+
+/-- Get the multiplicative → additive translation for the given name. -/
+def findTranslation? (env : Environment) : Name → Option Name :=
+  (ToAdditive.translations.getState env).find?
+
+def insertTranslation (src tgt : Name) : CoreM Unit := do
+  if let some tgt' := findTranslation? (←getEnv) src then
+    throwError "Already exists translation {src} ↦ {tgt'}"
+  (ToAdditive.translations.addEntry (←getEnv)) (src, tgt) |> setEnv
+  trace[to_additive] "Added translation {src} ↦ {tgt}."
+
+/-- Get whether or not the replace-all flag is set. -/
+def replaceAll [Functor M] [MonadOptions M] : M Bool :=
```

Comment:
```suggestion
/-- Get whether or not the detail-all flag is set. -/
def detailAll [Functor M] [MonadOptions M] : M Bool :=
```
?

Reply:
replaceAll is what it is called in mathlib3, so I think this is right, I've added a docstring.